dotCMS/src/main/java/com/dotmarketing/util/PageMode.java

```diff
@@ -0,0 +1,73 @@
+package com.dotmarketing.util;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+/**
+ * Usage:
+ * 
+ * PageMode mode = PageMode.get(request);
+ * 
+ * mode.isAdmin ; mode.showLive ;
+ * 
+ * or
+ * 
+ * if( PageMode.get(request).isAdmin){ doAdminStuff(); }
+ * 
+ * @author will
+ *
+ */
+public enum PageMode {
+
+    ANON(true, false), 
+    LIVE(true, true), 
+    PREVIEW(false, true), 
+    EDIT(false, true);
+
+    public final boolean showLive;
+    public final boolean isAdmin;
+
+    PageMode(boolean live, boolean admin) {
+        this.showLive = live;
+        this.isAdmin = admin;
+    }
+
+
+    public static PageMode get(final HttpSession ses) {
+
+        PageMode mode = (ses != null && ses.getAttribute(com.dotmarketing.util.WebKeys.PAGE_MODE_SESSION) != null
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "mode". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)